<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="Some bioinformatics knowledge gathered by Jacques Dainat"><meta name=author content="Jacques Dainat"><link href=https://juke34.github.io/knowledge/bioinformatics/format/handling/split_fasta/ rel=canonical><link href=../../conversion/gff_to_bed/ rel=prev><link href=../../../annotation/structural/annotation_tools_genome/ rel=next><link rel=icon href=../../../../assets/images/favicon.png><meta name=generator content="mkdocs-1.6.1, mkdocs-material-9.7.1"><title>FASTA - Jacques Dainat Bioinformatics Knowledge</title><link rel=stylesheet href=../../../../assets/stylesheets/main.484c7ddc.min.css><link rel=stylesheet href=../../../../assets/stylesheets/palette.ab4e12ef.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><link rel=stylesheet href=../../../../stylesheets/extra.css><script>__md_scope=new URL("../../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary=indigo data-md-color-accent=indigo> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#split-a-fasta-file class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../../../.. title="Jacques Dainat Bioinformatics Knowledge" class="md-header__button md-logo" aria-label="Jacques Dainat Bioinformatics Knowledge" data-md-component=logo> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Jacques Dainat Bioinformatics Knowledge </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> FASTA </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=default data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to dark mode" type=radio name=__palette id=__palette_0> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to light mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_0 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg> </label> </form> <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" title=Share aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg> </a> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap tabindex=0 data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/juke34/knowledge title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg> </div> <div class=md-source__repository> knowledge </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../../../.. class=md-tabs__link> Home </a> </li> <li class="md-tabs__item md-tabs__item--active"> <a href=../../specification/bam_and_sam/ class=md-tabs__link> Bioinformatics </a> </li> <li class=md-tabs__item> <a href=../../../../cluster/scheduler_rosetta/ class=md-tabs__link> Cluster </a> </li> <li class=md-tabs__item> <a href=../../../../programming/bash/bash_lecture/ class=md-tabs__link> Programming </a> </li> <li class=md-tabs__item> <a href=../../../../resource/course/ class=md-tabs__link> Resource </a> </li> <li class=md-tabs__item> <a href=../../../../sequencing/sequencing/ class=md-tabs__link> Sequencing </a> </li> <li class=md-tabs__item> <a href=../../../../statistics/statistics/ class=md-tabs__link> Statistics </a> </li> </ul> </div> </nav> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=../../../.. title="Jacques Dainat Bioinformatics Knowledge" class="md-nav__button md-logo" aria-label="Jacques Dainat Bioinformatics Knowledge" data-md-component=logo> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg> </a> Jacques Dainat Bioinformatics Knowledge </label> <div class=md-nav__source> <a href=https://github.com/juke34/knowledge title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg> </div> <div class=md-source__repository> knowledge </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../../.. class=md-nav__link> <span class=md-ellipsis> Home </span> </a> </li> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2 checked> <label class=md-nav__link for=__nav_2 id=__nav_2_label tabindex> <span class=md-ellipsis> Bioinformatics </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_2_label aria-expanded=true> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Bioinformatics </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_1 checked> <label class=md-nav__link for=__nav_2_1 id=__nav_2_1_label tabindex=0> <span class=md-ellipsis> Formats </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=2 aria-labelledby=__nav_2_1_label aria-expanded=true> <label class=md-nav__title for=__nav_2_1> <span class="md-nav__icon md-icon"></span> Formats </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_1_1> <label class=md-nav__link for=__nav_2_1_1 id=__nav_2_1_1_label tabindex=0> <span class=md-ellipsis> Specification </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=3 aria-labelledby=__nav_2_1_1_label aria-expanded=false> <label class=md-nav__title for=__nav_2_1_1> <span class="md-nav__icon md-icon"></span> Specification </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../specification/bam_and_sam/ class=md-nav__link> <span class=md-ellipsis> BAM / SAM </span> </a> </li> <li class=md-nav__item> <a href=../../specification/bed/ class=md-nav__link> <span class=md-ellipsis> BED v1 </span> </a> </li> <li class=md-nav__item> <a href=../../specification/cigar/ class=md-nav__link> <span class=md-ellipsis> CIGAR </span> </a> </li> <li class=md-nav__item> <a href=../../specification/cram/ class=md-nav__link> <span class=md-ellipsis> CRAM v3.1 </span> </a> </li> <li class=md-nav__item> <a href=../../specification/gxf/ class=md-nav__link> <span class=md-ellipsis> GFF / GTF all versions </span> </a> </li> <li class=md-nav__item> <a href=../../specification/markdown/ class=md-nav__link> <span class=md-ellipsis> Markdown </span> </a> </li> <li class=md-nav__item> <a href=../../specification/sugar/ class=md-nav__link> <span class=md-ellipsis> SUGAR </span> </a> </li> <li class=md-nav__item> <a href=../../specification/textile/ class=md-nav__link> <span class=md-ellipsis> TEXTILE (Wiki) </span> </a> </li> <li class=md-nav__item> <a href=../../specification/vcf/ class=md-nav__link> <span class=md-ellipsis> VCF v4.4 </span> </a> </li> <li class=md-nav__item> <a href=../../specification/vulgar/ class=md-nav__link> <span class=md-ellipsis> VULGAR </span> </a> </li> <li class=md-nav__item> <a href=../../specification/interesting_ressources/ class=md-nav__link> <span class=md-ellipsis> Interesting ressources about formats </span> </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_1_2> <label class=md-nav__link for=__nav_2_1_2 id=__nav_2_1_2_label tabindex=0> <span class=md-ellipsis> Conversion </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=3 aria-labelledby=__nav_2_1_2_label aria-expanded=false> <label class=md-nav__title for=__nav_2_1_2> <span class="md-nav__icon md-icon"></span> Conversion </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_1_2_1> <label class=md-nav__link for=__nav_2_1_2_1 id=__nav_2_1_2_1_label tabindex=0> <span class=md-ellipsis> GFF to </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=4 aria-labelledby=__nav_2_1_2_1_label aria-expanded=false> <label class=md-nav__title for=__nav_2_1_2_1> <span class="md-nav__icon md-icon"></span> GFF to </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../conversion/gff_to_gtf/ class=md-nav__link> <span class=md-ellipsis> GTF </span> </a> </li> <li class=md-nav__item> <a href=../../conversion/gff_to_bed/ class=md-nav__link> <span class=md-ellipsis> BED </span> </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_1_3 checked> <label class=md-nav__link for=__nav_2_1_3 id=__nav_2_1_3_label tabindex=0> <span class=md-ellipsis> Handling </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=3 aria-labelledby=__nav_2_1_3_label aria-expanded=true> <label class=md-nav__title for=__nav_2_1_3> <span class="md-nav__icon md-icon"></span> Handling </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> <span class=md-ellipsis> FASTA </span> <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> <span class=md-ellipsis> FASTA </span> </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#review-of-the-main-ways-to-split-a-fasta-file class=md-nav__link> <span class=md-ellipsis> Review of the main ways to split a FASTA file </span> </a> </li> </ul> </nav> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2_2> <label class=md-nav__link for=__nav_2_2 id=__nav_2_2_label tabindex=0> <span class=md-ellipsis> Annotation </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=2 aria-labelledby=__nav_2_2_label aria-expanded=false> <label class=md-nav__title for=__nav_2_2> <span class="md-nav__icon md-icon"></span> Annotation </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../../annotation/structural/annotation_tools_genome/ class=md-nav__link> <span class=md-ellipsis> Genome </span> </a> </li> <li class=md-nav__item> <a href=../../../annotation/structural/annotation_tools_mitome/ class=md-nav__link> <span class=md-ellipsis> Mitochondrion </span> </a> </li> <li class=md-nav__item> <a href=../../../annotation/structural/annotation_tools_plastome/ class=md-nav__link> <span class=md-ellipsis> Chloroplast </span> </a> </li> <li class=md-nav__item> <a href=../../../annotation/structural/annotation_tools_plasmidome/ class=md-nav__link> <span class=md-ellipsis> Plasmid </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../../taxonomic_classification/ class=md-nav__link> <span class=md-ellipsis> Classification </span> </a> </li> <li class=md-nav__item> <a href=../../../variant_annotation/ class=md-nav__link> <span class=md-ellipsis> Variant annotation </span> </a> </li> <li class=md-nav__item> <a href=../../../data_visualisation/ class=md-nav__link> <span class=md-ellipsis> Visualisation </span> </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_3> <label class=md-nav__link for=__nav_3 id=__nav_3_label tabindex=0> <span class=md-ellipsis> Cluster </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_3_label aria-expanded=false> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Cluster </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../../../cluster/scheduler_rosetta/ class=md-nav__link> <span class=md-ellipsis> Schedulers commands comparison </span> </a> </li> <li class=md-nav__item> <a href=../../../../cluster/slurm/ class=md-nav__link> <span class=md-ellipsis> Slurm </span> </a> </li> <li class=md-nav__item> <a href=../../../../cluster/rackham/ class=md-nav__link> <span class=md-ellipsis> Rackham </span> </a> </li> <li class=md-nav__item> <a href=../../../../cluster/screen/ class=md-nav__link> <span class=md-ellipsis> Screen </span> </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_4> <label class=md-nav__link for=__nav_4 id=__nav_4_label tabindex=0> <span class=md-ellipsis> Programming </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_4_label aria-expanded=false> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Programming </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_4_1> <label class=md-nav__link for=__nav_4_1 id=__nav_4_1_label tabindex=0> <span class=md-ellipsis> Bash </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=2 aria-labelledby=__nav_4_1_label aria-expanded=false> <label class=md-nav__title for=__nav_4_1> <span class="md-nav__icon md-icon"></span> Bash </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../../../programming/bash/bash_lecture/ class=md-nav__link> <span class=md-ellipsis> Lecture </span> </a> </li> <li class=md-nav__item> <a href=../../../../programming/bash/bash_cheat_sheet/ class=md-nav__link> <span class=md-ellipsis> Cheat sheet </span> </a> </li> <li class=md-nav__item> <a href=../../../../programming/bash/bash_other/ class=md-nav__link> <span class=md-ellipsis> Useful commands </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../../../programming/bioconda/bioconda/ class=md-nav__link> <span class=md-ellipsis> Bioconda </span> </a> </li> <li class=md-nav__item> <a href=../../../../programming/git/git/ class=md-nav__link> <span class=md-ellipsis> Git </span> </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_4_4> <label class=md-nav__link for=__nav_4_4 id=__nav_4_4_label tabindex=0> <span class=md-ellipsis> Perl </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=2 aria-labelledby=__nav_4_4_label aria-expanded=false> <label class=md-nav__title for=__nav_4_4> <span class="md-nav__icon md-icon"></span> Perl </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../../../programming/perl/perl_code_profiler/ class=md-nav__link> <span class=md-ellipsis> Perl code profiler </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../../../programming/python/python/ class=md-nav__link> <span class=md-ellipsis> Python </span> </a> </li> <li class=md-nav__item> <a href=../../../../programming/r/r/ class=md-nav__link> <span class=md-ellipsis> R </span> </a> </li> <li class=md-nav__item> <a href=../../../../programming/nextflow/nextflow/ class=md-nav__link> <span class=md-ellipsis> Nextflow </span> </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_5> <label class=md-nav__link for=__nav_5 id=__nav_5_label tabindex=0> <span class=md-ellipsis> Resource </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_5_label aria-expanded=false> <label class=md-nav__title for=__nav_5> <span class="md-nav__icon md-icon"></span> Resource </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../../../resource/course/ class=md-nav__link> <span class=md-ellipsis> Course </span> </a> </li> <li class=md-nav__item> <a href=../../../../resource/sites/ class=md-nav__link> <span class=md-ellipsis> Sites </span> </a> </li> <li class=md-nav__item> <a href=../../../../resource/blogs/ class=md-nav__link> <span class=md-ellipsis> Blogs </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../../../sequencing/sequencing/ class=md-nav__link> <span class=md-ellipsis> Sequencing </span> </a> </li> <li class=md-nav__item> <a href=../../../../statistics/statistics/ class=md-nav__link> <span class=md-ellipsis> Statistics </span> </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/juke34/knowledge/edit/main/docs/bioinformatics/format/handling/split_fasta.md title="Edit this page" class="md-content__button md-icon" rel=edit> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4zm10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1z"/></svg> </a> <a href=https://github.com/juke34/knowledge/raw/main/docs/bioinformatics/format/handling/split_fasta.md title="View source of this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.2 8.2 0 0 1-1.23-2"/></svg> </a> <h1 id=split-a-fasta-file>Split a FASTA file<a class=headerlink href=#split-a-fasta-file title="Permanent link">#</a></h1> <h2 id=review-of-the-main-ways-to-split-a-fasta-file>Review of the main ways to split a FASTA file<a class=headerlink href=#review-of-the-main-ways-to-split-a-fasta-file title="Permanent link">#</a></h2> <table> <thead> <tr> <th>tool</th> <th>language</th> <th>One sequence per file</th> <th>Can select nb of output files</th> <th>Can select nb seq by file</th> <th>Can select size of output files</th> <th>Overlap possible (when sequence cut)</th> <th>Can cut sequences</th> <th>Subsample possible</th> <th>Example</th> <th>Comment</th> </tr> </thead> <tbody> <tr> <td>awk</td> <td>awk</td> <td>yes</td> <td>no</td> <td>yes</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td><a href=#awk>example</a></td> <td></td> </tr> <tr> <td>split</td> <td>bash</td> <td>yes</td> <td>no</td> <td>yes</td> <td>yes</td> <td>no</td> <td>no</td> <td>no</td> <td><a href=#split>example</a></td> <td>Fasta must be single line fasta (one header + one single sequence line)</td> </tr> <tr> <td>bash</td> <td>bash</td> <td>yes</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td><a href=#bash>example</a></td> <td>Individual files will have the name of the corresponding sequence, without leading &gt;</td> </tr> <tr> <td>gaas_fasta_splitter.pl from <a href=https://github.com/NBISweden/GAAS>GAAS</a></td> <td>Perl</td> <td>yes</td> <td>yes</td> <td>yes</td> <td>no</td> <td>yes</td> <td>yes</td> <td>yes (stop when nb of files with the nb of seq asked reached)</td> <td><a href=#agat>example</a></td> <td></td> </tr> <tr> <td><a href=https://pypi.org/project/pyfasta/#command-line-interface>PyFasta</a></td> <td>Python</td> <td>yes</td> <td>yes</td> <td>no</td> <td>no</td> <td>yes</td> <td>yes</td> <td>NA</td> <td><a href=#pyfasta>example</a></td> <td></td> </tr> <tr> <td><a href=https://github.com/mdshw5/pyfaidx>pyfaidx</a></td> <td>Python</td> <td>yes</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td><a href=#pyfaidx>example</a></td> <td></td> </tr> <tr> <td><a href=https://github.com/genometools/genometools>GenomeTools</a></td> <td>Mostly C</td> <td>yes</td> <td>yes</td> <td>no</td> <td>yes</td> <td>no</td> <td>no</td> <td>no</td> <td><a href=#genometools>example</a></td> <td></td> </tr> <tr> <td><a href=http://emboss.sourceforge.net/apps/release/6.6/emboss/apps/seqretsplit.html>seqretsplit</a> from <a href=http://emboss.sourceforge.net/what/ >EMBOSS</a></td> <td>C</td> <td>yes</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td><a href=#emboss>example</a></td> <td></td> </tr> <tr> <td>bp_seqretsplit.pl from <a href=https://github.com/bioperl/bioperl-live>Bioperl</a></td> <td>perl</td> <td>yes</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td><a href=#bp_seqretsplit>example</a></td> <td></td> </tr> <tr> <td>faSplit from <a href=http://hgdownload.cse.ucsc.edu/admin/exe/ >Kent utils</a></td> <td>C</td> <td>yes</td> <td>yes</td> <td>no</td> <td>yes</td> <td>yes</td> <td>yes</td> <td>no</td> <td><a href=#faSplit>example</a></td> <td></td> </tr> <tr> <td>partition.sh from <a href=https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbmap-guide/ >BBMap</a></td> <td>Java</td> <td>no</td> <td>yes</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td>no</td> <td><a href=#bbmap>example</a></td> <td>multithreaded</td> </tr> <tr> <td><a href=https://github.com/shenwei356/seqkit>seqkit</a></td> <td>Go</td> <td>yes</td> <td>yes</td> <td>yes</td> <td>no</td> <td>no</td> <td>no</td> <td>yes (subsequence of given region)</td> <td><a href=https://bioinf.shenwei.me/seqkit/usage/#split>example</a></td> <td></td> </tr> <tr> <td><a href=https://github.com/sing-group/seda>SEDA</a></td> <td>java</td> <td>yes</td> <td>yes</td> <td>yes</td> <td>no</td> <td>no</td> <td>no</td> <td>yes (randomizable)</td> <td><a href=https://www.sing-group.org/seda/manual/operations.html#id19>example</a></td> <td>GUI only. Using <code>Independent extractions</code> and <code>Randomize</code> options give the possibility to get sequences picked several times. There is an extra function called <a href=https://www.sing-group.org/seda/manual/operations.html#regular-expression-split>regular expression split</a> (use of regex for selecting sequence by matching headers)</td> </tr> </tbody> </table> <h1 id=example>Example<a class=headerlink href=#example title="Permanent link">#</a></h1> <h2 id=awk>Awk<a class=headerlink href=#awk title="Permanent link">#</a></h2> <p>size = chunk size pre = output file prefix pad = padding width (the width of the numeric suffix).</p> <div class=highlight><pre><span></span><code>awk -v size=1000 -v pre=prefix -v pad=5 &#39;
   /^&gt;/ { n++; if (n % size == 1) { close(fname); fname = sprintf(&quot;%s.%0&quot; pad &quot;d&quot;, pre, n) } }
   { print &gt;&gt; fname }
&#39; input.fasta
</code></pre></div> <h2 id=split>Split<a class=headerlink href=#split title="Permanent link">#</a></h2> <p><code>split -l 2000 input.fasta</code></p> <h2 id=bash>Bash<a class=headerlink href=#bash title="Permanent link">#</a></h2> <div class=highlight><pre><span></span><code>while read line
do
    if [[ ${line:0:1} == &#39;&gt;&#39; ]]
    then
        outfile=${line#&gt;}.fa
        echo $line &gt; $outfile
    else
        echo $line &gt;&gt; $outfile
    fi
done &lt; myseq.fa
</code></pre></div> <h2 id=gaas>GAAS<a class=headerlink href=#gaas title="Permanent link">#</a></h2> <p>split the fasta file into one file per sequence</p> <p><code>gaas_fasta_splitter.pl -f input.fa --nb_seq_by_chunk 1</code></p> <p>split the fasta file into files of 100 sequences</p> <p><code>gaas_fasta_splitter.pl --nb_seq_by_chunk 100</code></p> <p>split the fasta file into 10 files</p> <p><code>gaas_fasta_splitter.pl --nb_chunks 10</code></p> <p>split the fasta file into 10 files and cut the sequence in chunk of 1000000 bp</p> <p><code>gaas_fasta_splitter.pl --nb_chunks 10 --size_seq 1000000</code></p> <p>split the fasta file into 10 files and cut the sequence in chunk of 1000000 bp with overlap of 2000 bp</p> <p><code>gaas_fasta_splitter.pl --nb_chunks 10 --size_seq 1000000 --overlap 2000</code></p> <p>split the fasta file into 10 files of 20 sequences and the original sequences are cut in chunk of 1000000 bp with overlap of 2000 bp. If all the input data cannot be contained into the 10 files of 20 sequences, the output is actually a subsample of the input data.</p> <p><code>gaas_fasta_splitter.pl --nb_chunks 10 --nb_seq_by_chunk 20 --size_seq 1000000 --overlap 2000</code></p> <h2 id=pyfasta>PyFasta<a class=headerlink href=#pyfasta title="Permanent link">#</a></h2> <p>split a fasta file into 6 new files of relatively even size:</p> <p><code>pyfasta split -n 6 original.fasta</code></p> <p>split the fasta file into one new file per header with “%(seqid)s” being filled into each filename.:</p> <p><code>pyfasta split –header “%(seqid)s.fasta” original.fasta</code></p> <p>create 1 new fasta file with the sequence split into 10K-mers:</p> <p><code>pyfasta split -n 1 -k 10000 original.fasta</code></p> <p>2 new fasta files with the sequence split into 10K-mers with 2K overlap:</p> <p><code>pyfasta split -n 2 -k 10000 -o 2000 original.fasta</code></p> <h2 id=pyfaidx>pyfaidx<a class=headerlink href=#pyfaidx title="Permanent link">#</a></h2> <p><code>faidx --split-files original.fasta</code></p> <h2 id=genometools>GenomeTools<a class=headerlink href=#genometools title="Permanent link">#</a></h2> <p><code>gt splitfasta -splitdesc multifastafile.fa</code></p> <h2 id=emboss>EMBOSS<a class=headerlink href=#emboss title="Permanent link">#</a></h2> <p><code>seqretsplit input.fa</code></p> <h2 id=bp_seqretsplit>bp_seqretsplit<a class=headerlink href=#bp_seqretsplit title="Permanent link">#</a></h2> <p><code>bp_seqretsplit file1 file2</code></p> <p>Similar to:<br> <div class=highlight><pre><span></span><code>#!/usr/bin/env perl

use strict;
use warnings;
use Bio::SeqIO;
my $in = Bio::SeqIO-&gt;new(-format =&gt; &#39;fasta&#39;,
                         -fh   =&gt; \*ARGV);
while( my $s = $in-&gt;next_seq ) {
    my ($id) = ($s-&gt;id =~ /^(?:\w+)\|(\S+)\|/);
    Bio::SeqIO-&gt;new(-format =&gt; &#39;fasta&#39;,
                    -file   =&gt; &quot;&gt;&quot;.$id.&quot;.fasta&quot;)-&gt;write_seq($s);
}
</code></pre></div></p> <h2 id=fasplit>faSplit<a class=headerlink href=#fasplit title="Permanent link">#</a></h2> <p>Break up scaffolds.fa using sequence names as file names (one sequence per file). Use the terminating / on the outRoot to get it to work correctly:</p> <p><code>faSplit byname scaffolds.fa outRoot/</code></p> <p>break up estAll.fa into 100 files (numbered est001.fa est002.fa, ... est100.fa Files will only be broken at fa record boundaries:</p> <p><code>faSplit sequence estAll.fa 100 est</code></p> <p>break up chr1.fa into 10 files:</p> <p><code>faSplit base chr1.fa 10 1_</code></p> <p>break up input.fa into 2000 base chunks:</p> <p><code>faSplit size input.fa 2000 outRoot</code></p> <p>break up est.fa into files of about 20000 bytes each by record:</p> <p><code>faSplit about est.fa 20000 outRoot</code></p> <p>Break up chrN.fa into files of at most 20000 bases each, at gap boundaries if possible. If the sequence ends in N's, the last piece, if larger than 20000, will be all one piece:</p> <p><code>faSplit gap chrN.fa 20000 outRoot</code></p> <h2 id=bbmap>BBMap<a class=headerlink href=#bbmap title="Permanent link">#</a></h2> <p>Split the fasta file into 5 files:</p> <p><code>partition.sh in=file.fasta out=part%.fasta ways=5</code></p> <h2 id=reference>Reference<a class=headerlink href=#reference title="Permanent link">#</a></h2> <p>https://www.biostars.org/p/229441/</p> </article> </div> <script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script> </div> <button type=button class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg> Back to top </button> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../../conversion/gff_to_bed/ class="md-footer__link md-footer__link--prev" aria-label="Previous: BED"> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg> </div> <div class=md-footer__title> <span class=md-footer__direction> Previous </span> <div class=md-ellipsis> BED </div> </div> </a> <a href=../../../annotation/structural/annotation_tools_genome/ class="md-footer__link md-footer__link--next" aria-label="Next: Genome"> <div class=md-footer__title> <span class=md-footer__direction> Next </span> <div class=md-ellipsis> Genome </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"annotate": null, "base": "../../../..", "features": ["announce.dismiss", "content.action.edit", "content.action.view", "content.code.annotate", "content.code.copy", "content.tooltips", "header.autohide", "navigation.footer", "navigation.indexes", "navigation.tabs", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow", "toc.integrate"], "search": "../../../../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script> <script src=../../../../assets/javascripts/bundle.79ae519e.min.js></script> <script src=https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js></script> <script src=../../../../javascripts/tablesort.js></script> </body> </html>